core/src/main/java/de/danoeh/antennapod/core/feed/FeedItemFilter.java

```diff
@@ -78,6 +80,45 @@ public FeedItemFilter(String[] properties) {
                     break;
             }
         }
+
+        mQuery = makeQuery();
+    }
+
+    private String makeQuery() {
+        // The keys used within this method, but explicitly combined with their table
+        String keyRead = PodDBAdapter.TABLE_NAME_FEED_ITEMS + "." + PodDBAdapter.KEY_READ;
+        String keyPosition = PodDBAdapter.TABLE_NAME_FEED_MEDIA + "." + PodDBAdapter.KEY_POSITION;
+        String keyDownloaded = PodDBAdapter.TABLE_NAME_FEED_MEDIA + "." + PodDBAdapter.KEY_DOWNLOADED;
+        String keyMediaId = PodDBAdapter.TABLE_NAME_FEED_MEDIA + "." + PodDBAdapter.KEY_ID;
+        String keyItemId = PodDBAdapter.TABLE_NAME_FEED_ITEMS + "." + PodDBAdapter.KEY_ID;
+        String keyFeedItem = PodDBAdapter.KEY_FEEDITEM;
+        String tableQueue = PodDBAdapter.TABLE_NAME_QUEUE;
+        String tableFavorites = PodDBAdapter.TABLE_NAME_FAVORITES;
+
+        List<String> statements = new ArrayList<>();
+        if (showPlayed)        statements.add(keyRead + " = 1 ");
```

Comment:
> I can conform to the Checkstyle rules if desired but i really think that the cascade of if statements profits in readability if each of them fits onto one line.

Readability was probably the cause of the [goto fail bug](https://en.wikipedia.org/wiki/Unreachable_code#goto_fail_bug), too. I would prefer to always use braces, even if it makes the code a bit longer.